src/Knp/Component/Pager/Event/Subscriber/Paginate/Doctrine/ODM/MongoDB/QuerySubscriber.php

```diff
@@ -36,8 +36,8 @@ public function items(ItemsEvent $event)
 
             $event->items = array();
             // iterator_to_array for GridFS results in 1 item
-            foreach ($cursor as $item) {
-                $event->items[] = $item;
+            while ($cursor->hasNext()) {
```

Comment:
What does justify this? Could you give more details please? Because, from [what I see](http://php.net/manual/en/mongocursor.getnext.php), the methods you used are just aliases to the standards `Iterator` methods. Unless I missed something, it doesn't make sense to do this.